packages/core/src/external/aws/lambda-logic/bundle-to-html-derm.ts

```diff
@@ -0,0 +1,2688 @@
+  });
+
+  const removeDuplicate = uniqWith(observationsSortedByDate, (a, b) => {
+    const aDate = dayjs(a.effectiveDateTime).format(ISO_DATE);
+    const bDate = dayjs(b.effectiveDateTime).format(ISO_DATE);
+    const aText = a.code?.text;
+    const bText = b.code?.text;
+    if (aText === undefined || bText === undefined) {
+      return false;
+    }
+    return aDate === bDate && aText === bText;
+  });
+
+  if (removeDuplicate.length === 0) {
+    return {
+      section: createChartSection(
+        `${vitalTitle} History`,
+        `<table><tbody><tr><td>No ${vitalTitle} readings found</td></tr></tbody></table>`,
+        false
+      ),
+      chartData: { labels: [], data: [] },
+    };
+  }
+  const { tableContent, chartData } = createVitalsChartByDate(removeDuplicate);
+
+  return {
+    section: createChartSection(`${vitalTitle} History`, tableContent, true),
+    chartData: chartData,
+  };
+}
+
+function createVitalsChartByDate(observations: Observation[]): {
+  tableContent: string;
+  chartData: ChartData;
+} {
+  const filteredObservations = filterObservationsByDate(observations);
+
+  const observationObjects: ObsSummary[] = filteredObservations
+    .flatMap(tables => {
+      return tables.observations.map(observation => {
+        const value = renderVitalsValue(observation);
+        if (value) {
+          return {
+            effectiveDate: dayjs(observation.effectiveDateTime).format(ISO_DATE),
+            vitalsValue: value,
+          };
+        }
+        return [];
+      });
+    })
+    .flat();
+
+  const observationsAscending = cloneDeep(observationObjects).sort((a, b) => {
+    return dayjs(a.effectiveDate).isBefore(dayjs(b.effectiveDate)) ? -1 : 1;
+  });
+
+  const chartData = {
+    labels: observationsAscending.map(obs => obs.effectiveDate),
+    data: observationsAscending.map(obs => parseFloat(obs.vitalsValue)),
+    min: Math.floor(Math.min(...observationsAscending.map(obs => parseFloat(obs.vitalsValue))) - 1),
+    max: Math.ceil(Math.max(...observationsAscending.map(obs => parseFloat(obs.vitalsValue))) + 1),
+  };
+
+  const observationRows = observationObjects
+    .map(obs => {
+      return `
+      <tr>
+      <td>${obs.effectiveDate}</td>
+      <td>${obs.vitalsValue}</td>
+      </tr>
+      `;
+    })
+    .join(" ");
+
+  const observationTableContents = `
+    <table>
+    <thead>
+    <tr>
+    <th style="width: 50%">Date</th>
+    <th style="width: 50%">Value</th>
+    </tr>
+    </thead>
+    <tbody>
+    ${observationRows}
+    </tbody>
+    </table>
+    `;
+
+  return { tableContent: observationTableContents, chartData };
+}
+
+function createChartSection(title: string, tableContents: string, contentPresent: boolean) {
+  return `
+      <div id="${title.toLowerCase().replace(/\s+/g, "-")}" class="section">
+        <div class="section-title">
+          <h3 id="${title}" title="${title}">&#x276F; ${title}</h3>
```

Comment:
Prob a good idea to come up with an ID that doesn't have spacing? I'd personally create an `id()` function that would take care of these details, and would pass context+value, in this example something like `id("chart-section-" + title)`

Reply:
Ill just simplify for now and just pass an explicit id

Reply:
Actually the id for the chart section isnt important for this summary so removing